lib/Backend/GlobOpt.cpp

```diff
@@ -8909,21 +8910,30 @@ GlobOpt::TypeSpecialization(
             {
                 return instr;
             }
+
+            // need to run typespec if cannot const fold InlineMathAbs, or dst will be dead
+            if (instr->m_opcode != Js::OpCode::InlineMathAbs)
+            {
+                runTypeSpec = false;
```

Comment:
That does not seem to be the problem at all.  Inlining of built-ins seems completely broken with -off:constfold.  Math.floor(1) is also broken.

All of this code:
        if (!this->IsLoopPrePass() &&
            !instr->IsBranchInstr() &&
            src1Val->GetValueInfo()->TryGetIntConstantValue(&intConstantValue) &&
            !(
                // Nothing to fold for element stores. Go into type specialization to see if they can at least be specialized.
                instr->m_opcode == Js::OpCode::StElemI_A ||
                instr->m_opcode == Js::OpCode::StElemI_A_Strict ||
                instr->m_opcode == Js::OpCode::StElemC ||
                instr->m_opcode == Js::OpCode::MultiBr ||
                instr->m_opcode == Js::OpCode::InlineArrayPop
            ))

Should really be in OptConstFoldUnary().  When OptConstFold returns false, we should go to the TypeSpecUnary() code.

---
In reply to: [99457671](https://github.com/Microsoft/ChakraCore/pull/2267#discussion_r99457671) [](ancestors = 99457671)